### PR TITLE
Payment

### PR DESCRIPTION
**Pull request by nathan-rees**
configured paypal api fully now

### CODE REVIEW

**Comment by RafSobolQAC** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5261906&bid=17238798)

File: app/controllers/PaymentController.scala [hunk 1]
```diff
@@ -35,28 +36,49 @@ class PaymentController @Inject()(ws:WSClient,cc: ControllerComponents) extends
...
   }
 
   def index = Action {
-   // Ok(views.html.index(getAccessToken()))
-    Ok(createOrder().toString)
-   //Ok(getAccessToken())
-    Ok(views.html.payment(69.00f))
+    //Ok(createOrder().toString())
+    val json=createOrder()
+    val url=(json\"links"\1\\"href").last.toString.replace("\"","")
```
Comment (RafSobolQAC) [inline app/controllers/PaymentController.scala]: Codacy Issue found: [Usage of last on collections.](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5261906)

File: app/controllers/PaymentController.scala [hunk 2]
```diff
@@ -35,28 +36,49 @@ class PaymentController @Inject()(ws:WSClient,cc: ControllerComponents) extends
...
+      post(json),Duration(5,"seconds")).json
+
+
+
+
+
   }
-  def lol: Unit ={
-    val request=ws.url("https://api.sandbox.paypal.com/v2/payments/authorizations/")
+  def capturePayment(orderID:String)=Action{
```
Comment (RafSobolQAC) [inline app/controllers/PaymentController.scala]: Codacy Issue found: [The method takes a value that is controlled by the client](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5261906)

File: app/controllers/PaymentController.scala [hunk 3]
```diff
@@ -35,28 +36,49 @@ class PaymentController @Inject()(ws:WSClient,cc: ControllerComponents) extends
...
 
   def index = Action {
-   // Ok(views.html.index(getAccessToken()))
-    Ok(createOrder().toString)
-   //Ok(getAccessToken())
-    Ok(views.html.payment(69.00f))
+    //Ok(createOrder().toString())
+    val json=createOrder()
+    val url=(json\"links"\1\\"href").last.toString.replace("\"","")
+    val orderID=(json\"id").get.toString.replace("\"","")
```
Comment (RafSobolQAC) [inline app/controllers/PaymentController.scala]: Codacy Issue found: [Usage of get on optional type.](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5261906)

**Comment by RafSobolQAC** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5261906&bid=17238798)

File: app/controllers/PaymentController.scala [hunk 4]
```diff
@@ -25,38 +26,58 @@ class PaymentController @Inject()(ws:WSClient,cc: ControllerComponents) extends
...
 
   def index = Action {
-   // Ok(views.html.index(getAccessToken()))
-    Ok(createOrder().toString)
-   //Ok(getAccessToken())
-    Ok(views.html.payment(69.00f))
+    //Ok(createOrder().toString())
+    val json=createOrder()
+    val url=(json\"links"\1\\"href").head.toString.replace("\"","")
+    val orderID=(json\"id").get.toString.replace("\"","")
```
Comment (RafSobolQAC) [inline app/controllers/PaymentController.scala]: Codacy Issue found: [Usage of get on optional type.](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5261906)

File: app/controllers/PaymentController.scala [hunk 5]
```diff
@@ -25,38 +26,58 @@ class PaymentController @Inject()(ws:WSClient,cc: ControllerComponents) extends
...
       addHttpHeaders("Content-Type"->"application/json").
       addHttpHeaders("Authorization"->token).
-      post(json)
-      request//this should be removed
+      post(json),Duration(5,"seconds")).json
+
   }
-  def lol: Unit ={
-    val request=ws.url("https://api.sandbox.paypal.com/v2/payments/authorizations/")
+  def capturePayment(orderID:String)=Action{
```
Comment (RafSobolQAC) [inline app/controllers/PaymentController.scala]: Codacy Issue found: [The method takes a value that is controlled by the client](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5261906)

File: app/controllers/PaymentController.scala [hunk 6]
```diff
@@ -25,38 +26,58 @@ class PaymentController @Inject()(ws:WSClient,cc: ControllerComponents) extends
...
   }
 
   def index = Action {
-   // Ok(views.html.index(getAccessToken()))
-    Ok(createOrder().toString)
-   //Ok(getAccessToken())
-    Ok(views.html.payment(69.00f))
+    //Ok(createOrder().toString())
+    val json=createOrder()
+    val url=(json\"links"\1\\"href").head.toString.replace("\"","")
```
Comment (RafSobolQAC) [inline app/controllers/PaymentController.scala]: Codacy Issue found: [Usage of head on collections.](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5261906)

**Comment by RafSobolQAC** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5261906&bid=17238798)

File: app/controllers/PaymentController.scala [hunk 7]
```diff
@@ -24,39 +25,52 @@ class PaymentController @Inject()(ws:WSClient,cc: ControllerComponents) extends
...
       addHttpHeaders("Authorization"->token).
-      post(json)
-      request//this should be removed
+      post(json),Duration(5,"seconds")).json
+
+
   }
-  def lol: Unit ={
-    val request=ws.url("https://api.sandbox.paypal.com/v2/payments/authorizations/")
+  def capturePayment(orderID:String)=Action{
```
Comment (RafSobolQAC) [inline app/controllers/PaymentController.scala]: Codacy Issue found: [The method takes a value that is controlled by the client](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5261906)

File: app/controllers/PaymentController.scala [hunk 8]
```diff
@@ -24,39 +25,52 @@ class PaymentController @Inject()(ws:WSClient,cc: ControllerComponents) extends
...
   }
 
   def index = Action {
-   // Ok(views.html.index(getAccessToken()))
-    Ok(createOrder().toString)
-   //Ok(getAccessToken())
-    Ok(views.html.payment(69.00f))
+    //Ok(createOrder().toString())
+    val json=createOrder()
+    val url=(json\"links"\1\\"href").head.toString.replace("\"","")
```
Comment (RafSobolQAC) [inline app/controllers/PaymentController.scala]: Codacy Issue found: [Usage of head on collections.](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5261906)

File: app/controllers/PaymentController.scala [hunk 9]
```diff
@@ -24,39 +25,52 @@ class PaymentController @Inject()(ws:WSClient,cc: ControllerComponents) extends
...
+    val orderID=(json\"id").get.toString.replace("\"","")
```
Comment (RafSobolQAC) [inline app/controllers/PaymentController.scala]: Codacy Issue found: [Usage of get on optional type.](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5261906)

File: app/controllers/PaymentController.scala [hunk 10]
```diff
@@ -24,39 +25,43 @@ class PaymentController @Inject()(ws:WSClient,cc: ControllerComponents) extends
...
-      addHttpHeaders("Content-Type"->"application/json").
-      addHttpHeaders("Authorization"->token).
-      post(json)
-      request//this should be removed
-  }
-  def lol: Unit ={
-    val request=ws.url("https://api.sandbox.paypal.com/v2/payments/authorizations/")
+    val json=createOrder(json2)
+    val url=(json\"links"\1\\"href")(0).toString.replace("\"","")
+    val orderID=(json\"id").get.toString.replace("\"","")
```
Comment (RafSobolQAC) [inline app/controllers/PaymentController.scala]: Codacy Issue found: [Usage of get on optional type.](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5261906)

**Comment by RafSobolQAC** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5261906&bid=17238798)

File: test/app/controllers/PayementControllerTest.scala [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+    "capture the current payment" in{
+      val result=controller.capturePayment("this is an order ID XD")
+      result.toString() must not be empty
+    }
+  }
+
+
+  "create Order" should {
+    "create an order" in new WithApplication() with Injecting {
+      val Action = inject[WSClient]
```
Comment (RafSobolQAC) [inline test/app/controllers/PayementControllerTest.scala]: Codacy Issue found: [Field name 'Action' does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5261906)

File: app/controllers/PaymentController.scala [hunk 12]
```diff
@@ -24,39 +25,43 @@ class PaymentController @Inject()(ws:WSClient,cc: ControllerComponents) extends
       val accessToken: String = (json \ "access_token").as[String]
       accessToken
     }
+  def createOrder(json:JsValue) ={
+    val token:String="Bearer "+getAccessToken()
+      Await.result(ws.url("https://api.sandbox.paypal.com/v2/checkout/orders").addHttpHeaders("Content-Type"->"application/json").addHttpHeaders("Authorization"->token).post(json),Duration(5,"seconds")).json
+  }
+  def capturePayment(orderID:String)=Action{
```
Comment (RafSobolQAC) [inline app/controllers/PaymentController.scala]: Codacy Issue found: [The method takes a value that is controlled by the client](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5261906)

**Comment by RafSobolQAC** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5261906&bid=17238798)

File: app/controllers/PaymentController.scala [hunk 13]
```diff
@@ -24,39 +25,40 @@ class PaymentController @Inject()(ws:WSClient,cc: ControllerComponents) extends
...
-      addHttpHeaders("Content-Type"->"application/json").
-      addHttpHeaders("Authorization"->token).
-      post(json)
-      request//this should be removed
-  }
-  def lol: Unit ={
-    val request=ws.url("https://api.sandbox.paypal.com/v2/payments/authorizations/")
+    val json=createOrder(json2)
+    val url=(json\"links"\1\\"href")(0).toString.replace("\"","")
+    val orderID=(json\"id").get.toString.replace("\"","")
```
Comment (RafSobolQAC) [inline app/controllers/PaymentController.scala:59]: Codacy Issue found: [Usage of get on optional type.](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5261906)

File: app/controllers/PaymentController.scala [hunk 14]
```diff
@@ -24,39 +25,40 @@ class PaymentController @Inject()(ws:WSClient,cc: ControllerComponents) extends
       val accessToken: String = (json \ "access_token").as[String]
       accessToken
     }
+  def createOrder(json:JsValue) ={
+    val token:String="Bearer "+getAccessToken()
+      Await.result(ws.url("https://api.sandbox.paypal.com/v2/checkout/orders").addHttpHeaders("Content-Type"->"application/json").addHttpHeaders("Authorization"->token).post(json),Duration(5,"seconds")).json
+  }
+  def capturePayment(orderID:String)=Action{
```
Comment (RafSobolQAC) [inline app/controllers/PaymentController.scala:32]: Codacy Issue found: [The method takes a value that is controlled by the client](https://app.codacy.com/manual/RafSobolQAC/cinema-team-one/pullRequest?prid=5261906)

**Review by johnpooler — APPROVED** [review]

**Pull request merged**